### ISSUE TITLE
InWhere EOF exception

### ISSUE THREAD

**Issue by include4eto** [phalcon/cphalcon#10676]
```php
$users = \Models\Users::query()
	->inWhere("id", array())
	->execute();
```

When `inWhere` is given an empty array, it crashes with the following exception:
```
Phalcon\Mvc\Model\Exception: Syntax error, unexpected EOF
```
 
At the very least, the error could make more sense :)

**Comment by andresgutierrez** [comment]
"id" is not an expression that can be evaluated so it encounters the EOF which is unexpected, totally make sense

**Comment by include4eto** [comment]
But it's not a syntax error.. Unexpected EOF has a completely different meaning.
It might make sense when you know how it works, but the exception is totally wrong.

**Comment by andrew-demb** [comment]
I'm sorry for up closed issue, but I have a question.
Why in \Phalcon\Mvc\Model\Criteria::inWhere() (https://github.com/phalcon/cphalcon/blob/cf3c1fe62184d36d24c543c0b4f9993ea8f2445b/phalcon/mvc/model/criteria.zep#L448) we don't check for empty array values and modify condition like the \Phalcon\Mvc\Model\Query\Builder::inWhere() (https://github.com/phalcon/cphalcon/blob/cf3c1fe62184d36d24c543c0b4f9993ea8f2445b/phalcon/mvc/model/query/builder.zep#L752)? I think it would have a bit more logic.
Phalcon 2.1.x

**Comment by sergeyklay** [comment]
@andrew-demb Could you please create separated issue with script to reproduce

**Comment by sergeyklay** [comment]
Fixed in the `3.0.x` branch.
